hyperspy/misc/slicing.py

```diff
@@ -186,9 +186,14 @@ def _get_array_slices(self, slices, isNavigation=None):
                 array_slices.append(slice_)
         return array_slices
 
-    def _slicer(self, slices, isNavigation=None):
+    def _slicer(self, slices, isNavigation=None, out=None):
         array_slices = self._get_array_slices(slices, isNavigation)
-        _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        if out is None:
+            _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        else:
+            _obj = out
+            _obj.axes_manager = self.axes_manager.deepcopy()
+            _obj.data = self._data[array_slices]  # Changes data id!
```

Comment:
The function is defined in the `interactive` branch, here: https://github.com/francisco-dlp/hyperspy/blob/NEW_Interactive/hyperspy/axes.py#L413-L437

My original code-block was just copy and paste. To be more precise, this:
```python
 _obj.axes_manager = self.axes_manager.deepcopy()
```
could be  replaced by this:
```python
for ax_dst, ax_src in zip(_obj.axes_manager._axes,
                          self.axes_manager._axes):
    ax_dst.update_from(ax_src, ('offset', 'scale', 'size'))
```